Makefile

```diff
@@ -51,11 +51,16 @@ assets:
 
 docker: build
 	@echo ">> building docker image"
-	@docker build -t "$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" .
+	@docker build -t improbable/"$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" .
+	@docker tag improbable/"$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" improbable/"$(DOCKER_IMAGE_NAME)":latest
+
+docker-push:
+	@echo ">> pushing image"
+	@docker push improbable/"$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)"
```

Comment:
hmm sorry for question after approval, but wonder if this logic  is actually the correct one:
1) Make build will  build `improbable/thanos:master` and tag this to `improbable/thanos:latest` as well
2) Push pushes  `improbable/thanos:master` only.

Hm.. what about pushing `latest` here maybe?
So maybe killing `tag` step and add `latest` as default `DOCKER_IMAGE_TAG` in Makefile. What do you think?

Reply:
@Bplotka just to clarify, are you saying the promu build step in `make build` would create the `improbable/thanos:latest` tag? If so I wasn't aware it was doing this, so yes I agree we can drop the tag step in `make docker`.

Reply:
No I mean your code here. We produce two tags but we push only one

Reply:
ahhh, right sorry for the confusion. I see, pushing latest definitely makes more sense for now. I actually think perhaps we should push both `DOCKER_IMAGE_TAG` and the `latest` tag! That way we can achieve something that looks like https://hub.docker.com/r/prom/prometheus/tags/ once git releases start getting tags against Thanos builds.

WDYT? Happy for me to go ahead an change `docker-push to look like this?
```
docker-push:
        @echo ">> pushing image"
        @docker push improbable/"$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)"
        @docker push improbable/"$(DOCKER_IMAGE_NAME):latest"
```

Reply:
So actually thanks for example. I would do exactly like there. There is a master tag, so let's push it here on every commit. But I would avoid latest tag here. So let's not push and tag latest in our circleCI. Latest should be recommended, working version. Does it make sense?

Reply:
How would we get the latest tag in dockerhub in that case? Manual push?

Reply:
Release push someday (:

Reply:
hehe. I guess.... I have wider thoughts on this but I'll raise in a separate PR. Lets get this in for now. I'll make the change to just push the DOCKER_IMAGE_TAG.